examples/api/lib/material/progress_indicator/circular_progress_indicator.2.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [CircularProgressIndicator] with [year2023] set to false.
+
+void main() => runApp(const ProgressIndicatorExampleApp());
+
+class ProgressIndicatorExampleApp extends StatelessWidget {
+  const ProgressIndicatorExampleApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: ProgressIndicatorExample(),
+    );
+  }
+}
+
+class ProgressIndicatorExample extends StatefulWidget {
+  const ProgressIndicatorExample({super.key});
+
+  @override
+  State<ProgressIndicatorExample> createState() =>
+      _ProgressIndicatorExampleState();
+}
+
+class _ProgressIndicatorExampleState extends State<ProgressIndicatorExample>
+    with TickerProviderStateMixin {
+  late AnimationController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+      vsync: this,
+      duration: const Duration(seconds: 5),
+    )..addListener(() {
+      setState(() {});
+    })
+    ..repeat(reverse: true);
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Center(
+        child: Padding(
+          padding: const EdgeInsets.all(20.0),
+          child: Column(
+            spacing: 16.0,
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              const Text('Determinate CircularProgressIndicator'),
+              CircularProgressIndicator(
+                year2023: false,
```

Comment:
I'm thinking whether we can create a ProgressIndicatorTheme object for year2023 appearance. We can also add a `Switch` to turn on/off the `year2023` flag. It can be useful when we remove this deprecated flag and users want to find a copy+paste way to still keep year2023 look.

Reply:
Sure, that would be good idea. I'll file separate PR for the theme flag.

Reply:
Updated the samples here with switch toggle for the year2023 flag and removed added samples.